sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,692 @@
+     * this RestProxy "implements".
+     */
+    private SyncRestProxy(HttpPipeline httpPipeline, SerializerAdapter serializer, SwaggerInterfaceParser interfaceParser) {
+        this.httpPipeline = httpPipeline;
+        this.serializer = serializer;
+        this.interfaceParser = interfaceParser;
+        this.decoder = new HttpResponseDecoder(this.serializer);
+    }
+
+    /**
+     * Get the SwaggerMethodParser for the provided method. The Method must exist on the Swagger interface that this
+     * RestProxy was created to "implement".
+     *
+     * @param method the method to get a SwaggerMethodParser for
+     * @return the SwaggerMethodParser for the provided method
+     */
+    private SwaggerMethodParser getMethodParser(Method method) {
+        return interfaceParser.getMethodParser(method);
+    }
+
+    /**
+     * Send the provided request asynchronously, applying any request policies provided to the HttpClient instance.
+     *
+     * @param request the HTTP request to send
+     * @param contextData the context
+     * @return a {@link Mono} that emits HttpResponse asynchronously
+     */
+    public HttpResponse send(HttpRequest request, Context contextData) {
+        return httpPipeline.sendSynchronously(request, contextData);
+    }
+
+    @Override
+    public Object invoke(Object proxy, final Method method, Object[] args) {
+        validateResumeOperationIsNotPresent(method);
+
+        try {
+            final SwaggerMethodParser methodParser = getMethodParser(method);
+            final HttpRequest request = createHttpRequest(methodParser, args);
+            Context context = methodParser.setContext(args);
+
+            RequestOptions options = methodParser.setRequestOptions(args);
+            context = mergeRequestOptionsContext(context, options);
+
+            context = context.addData("caller-method", methodParser.getFullyQualifiedMethodName())
+                .addData("azure-eagerly-read-response", shouldEagerlyReadResponse(methodParser.getReturnType()));
+            context = startTracingSpan(method, context);
+
+            // If there is 'RequestOptions' apply its request callback operations before validating the body.
+            // This is because the callbacks may mutate the request body.
+            if (options != null) {
+                options.getRequestCallback().accept(request);
+            }
+
+            if (request.getBody() != null) {
+                request.setBody(validateLength(request));
+            }
+
+            final HttpResponse response = send(request, context);
+
+            HttpDecodedResponse decodedResponse = this.decoder.decodeSync(response, methodParser);
+
+            return handleRestReturnType(decodedResponse, methodParser,
+                methodParser.getReturnType(), context, options);
+        } catch (IOException e) {
+            throw LOGGER.logExceptionAsError(Exceptions.propagate(e));
```

Comment:
let's not use `Exceptions.propagate` in any of sync call stacks. This is going to throw exception from reactor.
`UncheckedIOException` is better candidate here.